libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/client/EvcsClient.java

```diff
@@ -156,14 +167,27 @@ private URI getUri(String userId, List<EvcsVCState> vcStatesToQueryFor)
     private void checkResponseStatusCode(HttpResponse<String> evcsHttpResponse)
             throws EvcsServiceException {
         if (200 > evcsHttpResponse.statusCode() || evcsHttpResponse.statusCode() > 299) {
+            String responseMessage;
+            try {
+                Map<String, String> responseBody =
+                        OBJECT_MAPPER.readValue(evcsHttpResponse.body(), new TypeReference<>() {});
+                responseMessage =
+                        Optional.ofNullable(responseBody.get("message"))
+                                .orElse("Received no evcs response body.");
+            } catch (JsonProcessingException e) {
+                responseMessage = "Failed to parse evcs response body.";
+            }
             LOGGER.info(
                     LogHelper.buildLogMessage(
                                     ErrorResponse.RECEIVED_NON_200_RESPONSE_STATUS_CODE
                                             .getMessage())
-                            .with(LOG_STATUS_CODE.getFieldName(), evcsHttpResponse.statusCode()));
-            throw new EvcsServiceException(
-                    HTTPResponse.SC_SERVER_ERROR,
-                    ErrorResponse.RECEIVED_NON_200_RESPONSE_STATUS_CODE);
+                            .with(LOG_STATUS_CODE.getFieldName(), evcsHttpResponse.statusCode())
+                            .with(LOG_MESSAGE_DESCRIPTION.getFieldName(), responseMessage));
+            if (evcsHttpResponse.statusCode() != 404) {
+                throw new EvcsServiceException(
+                        HTTPResponse.SC_SERVER_ERROR,
+                        ErrorResponse.RECEIVED_NON_200_RESPONSE_STATUS_CODE);
+            }
         }
         LOGGER.info(LogHelper.buildLogMessage("Successful HTTP response from EVCS Api"));
```

Comment:
Do we still want this log message if it was a 404?

Reply:
Good to print log for any response code other then 2xx